module-new/src/components/textArea/textArea.component.tsx

```diff
@@ -0,0 +1,216 @@
+  value?: TValue;
+}
+
+const DebounceTextAreaBase = React.forwardRef<HTMLTextAreaElement, IDelayedTextAreaBaseProps<string>>(
+  ({ milliseconds, value, onValueChange, onChange, ...nativeProps }, ref) => {
+    const [actualValue, setActualValue] = useDebounce(milliseconds, value, onValueChange);
+
+    const onChangeEvent = React.useCallback(
+      (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        setActualValue(e.currentTarget.value);
+        onChange?.(e);
+      },
+      [setActualValue, onChange]
+    );
+
+    return <textarea ref={ref} value={actualValue} onChange={onChangeEvent} {...nativeProps} />;
+  }
+);
+
+DebounceTextAreaBase.displayName = 'DebounceInput';
+
+interface ITextAreaProps<TValue extends NullOrUndefined<string> | NullOrUndefined<number>>
+  extends Omit<IInputWrapperProps, 'onClick' | 'onValueChange'>,
+    NativeTextAreaProps {
+  /** A class name to apply to the input element */
+  textAreaClassName?: string;
+
+  /**  prop for binding to an Armstrong form binder (see forms documentation) */
+  bind?: IBindingProps<TValue>;
+
+  /** Called when the value changes, takes into account any delay values and other effects. */
+  onValueChange?: (value?: TValue) => void;
+
+  /** The delay config, used to set throttle and debounce values. */
+  delay?: number;
+
+  /** The current value of the input */
+  value?: TValue;
+
+  /** which size variant to use */
+  displaySize?: TextAreaDisplaySize;
+
+  /** A callback function to handle onChange event */
+  onChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
+
+  /** An ID for the label to use when testing  */
+  testId?: string;
+}
+
+/** A component which wraps up a native text area element with some binding logic, labels and validation errors. */
+export const TextArea = React.forwardRef<HTMLTextAreaElement, ITextAreaProps<string>>(
+  (
+    {
+      bind,
+      onChange,
+      value,
+      className,
+      validationErrorMessages,
+      validationMode,
+      pending,
+      disabled,
+      disableOnPending,
+      onValueChange,
+      scrollValidationErrorsIntoView,
+      delay,
+      validationErrorsClassName,
+      statusClassName,
+      textAreaClassName,
+      label,
+      required,
+      requiredIndicator,
+      displaySize,
+      labelClassName,
+      labelId,
+      testId,
+      ...nativeProps
+    },
+    ref
+  ) => {
+    const reactId = React.useId();
+    const id = nativeProps.id ?? reactId;
+
+    const globals = useArmstrongConfig({
+      validationMode,
+      disableInputOnPending: disableOnPending,
+      inputDisplaySize: displaySize,
+      requiredIndicator,
+      scrollValidationErrorsIntoView,
+    });
+
+    const [boundValue, setBoundValue, bindConfig] = useBindingState(bind, {
+      value: value?.toString(),
+      validationErrorMessages,
+      validationMode: 'message',
+    });
+
+    const parseValue = React.useCallback((unparsedValue?: string) => {
+      return unparsedValue;
+    }, []);
```

Comment:
not sure this function is doing anything?

Reply:
Ah yep, you right